resources/helm/dask-gateway/templates/scheduler-proxy-service.yaml

```diff
@@ -0,0 +1,32 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: scheduler-proxy-api
+  labels:
+    {{- include "dask-gateway.labels" . | nindent 4 }}
+    app.kubernetes.io/component: scheduler-proxy
+spec:
+  selector:
+    {{- include "dask-gateway.labels" . | nindent 4 }}
+    app.kubernetes.io/component: scheduler-proxy
+  ports:
+    - protocol: TCP
+      port: 8001
+      targetPort: 8001
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: scheduler-proxy-public
+  labels:
+    {{- include "dask-gateway.labels" . | nindent 4 }}
+    app.kubernetes.io/component: scheduler-proxy
+spec:
+  selector:
+    {{- include "dask-gateway.labels" . | nindent 4 }}
+    app.kubernetes.io/component: scheduler-proxy
+  ports:
+    - protocol: TCP
+      port: 8786
+      targetPort: 8786
+  type: LoadBalancer
```

Comment:
I would make this configurable. Load balancers can be expensive and people often consolidate those costs using ingresses.

Reply:
This service uses raw `tcp` and not `http`, which as I understand doesn't work with kubernetes ingress? The other proxy definitely could use an ingress.

Reply:
You're right that reverse proxies are layer 7 and therefore only support http/https. 

However some ingress controllers like nginx [do support tcp forwarding](https://kubernetes.github.io/ingress-nginx/user-guide/exposing-tcp-udp-services/). They actually avoid nginx all together and just add a new port endpoint to the existing loadbalancer and forward the traffic straight into your service. It requires a little more setup on the user's part, they would need to create an additional resource as well as install the helm chart, but they would need to be able to set the type to `ClusterIP` in the chart.

Reply:
[Looks like trafik also supports TCP 🎉.](https://github.com/containous/traefik/pull/4587)

Reply:
I've made the type configurable, defaulting to `LoadBalancer` (mirroring that done in zero-to-jupyterhub-k8s).